test-fixture/src/header_protection.rs

```diff
@@ -120,11 +120,9 @@ pub fn remove_header_protection(hp: &HpKey, header: &[u8], payload: &[u8]) -> (V
     // Trim down to size.
     fixed_header.truncate(pn_offset + pn_len);
     // The packet number should be 1.
-    let pn = Decoder::new(&fixed_header[pn_offset..])
-        .decode_uint(pn_len)
-        .unwrap();
-
-    (fixed_header, pn)
+    let mut pn = [0; 8];
+    pn[8 - pn_len..].copy_from_slice(&fixed_header[pn_offset..pn_offset + pn_len]);
```

Comment:
What is the reason for no longer going through `Decoder` here?

Reply:
I didn't want to expose the three-byte variant.  It seemed like an OK compromise given that we aren't using `Decoder` for its `Cursor`-like capabilities, just the integer decoding.

Reply:
Maybe make that reasoning a comment?